lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/IO.hs

```diff
@@ -211,6 +212,9 @@ getBIP32PathsForOwnedInputs
 getBIP32PathsForOwnedInputs a w =
     Wallet.getBIP32PathsForOwnedInputs a <$> readWalletState w
 
+signTxBody :: Write.TxBody -> [BIP32Path] -> WalletInstance -> IO Write.Tx
```

Comment:
```suggestion
signTxBody :: Write.TxBody -> WalletInstance -> IO (Maybe Write.Tx)
```

> But is returning IO (Maybe result) in the .IO module our desired approach then like

I think so 🤔 — The ubiquitous `ErrNoSuchWallet` is gone, and by making the failure explicitly in the result, we can also present it an explicit HTTP status code 4xx response. I would only sweep the status code 500 "internal server error" into the `IO` exceptions.